File: reuse_java/src/main/java/com/kmmania/Athlete.java
package com.kmmania;

/**
 * The Athlete class defines a profile for an athlete.
 *
 * @author Dac Linh Giao NGUYEN
 * @version 0.1
 */
public class Athlete {
    private final String speciality;

    /**
     * Constructs an Athlete object with the specified specialty.
     *
     * @param speciality the specialty of the athlete
     */
    public Athlete(String speciality) {
        this.speciality = speciality;
    }

    /**
     * Returns a string representation of the athlete's specialty.
     *
     * @return a string representing the athlete's specialty
     */
    public String mySpeciality() {
        return "My speciality is " + this.speciality + ".";
    }
}

File: reuse_js/src/athlete.js
/**
 * Represents an athlete.
 *
 * @author Dac Linh Giao NGUYEN
 * @version 0.1
 * @since 0.1
 */
class Athlete {
    /**
     * Create an athlete.
     *
     * @param {string} speciality - The speciality of the athlete.
     */
    constructor(speciality) {
        this.speciality = speciality;
    }

    /**
     * Get the speciality of the athlete.
     *
     * @returns {string} The speciality of the athlete.
     */
    mySpeciality() {
        return `My speciality is ${this.speciality}.`;
    }
}

module.exports = Athlete;